### PR TITLE
Fix bug #111 New field initiating_party_identifier on res.company

### PR DESCRIPTION
**Pull request by alexis-via**
migration script and post-install script provided

Special thanks to @lmignon  for his help on the development of this PR.

### CODE REVIEW

**Comment by lmignon** [comment]
@alexis-via with pleasure :smile:

File: account_banking_pain_base/post_install.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+
+def set_default_initiating_party(cr, registry):
+    party_issuer_per_country = {
+        'BE': 'KBO-BCE',  # KBO-BCE = the registry of companies in Belgium
+    }
+    cr.execute('''
```
Comment (pedrobaeza) [inline account_banking_pain_base/post_install.py]: You can avoid SQL using registry, that works like `self.pool`

**Comment by pedrobaeza** [comment]
Thanks for the PR. These are the holes I see:
- There's no guidance when creating a new company for composing this field. You can provide the calculation via onchange.
- There's no exception raise when issuing a DD and this field is not filled. I remind you that at least in Spain, this data is mandatory, and people send their files thinking that all it's correct, but the bank refuses them.

**Comment by alexis-via** [comment]
@pedrobaeza Here are my answers to your remarks:
The code in post_install.py is mutualized with the code of the migration script in which I don't have "registry" as argument, that's why I don't use it.
About the "holes":
1) with @sbidoul, we think that the on_change should be provided in country-specific modules in each OCA localization, because each country has it's own method for composing the field
2) you could inherit the field to make it a required field for spain in a spain-specific module

**Comment by pedrobaeza** [comment]
I see your reasoning. But you're not providing a specific hook for making that things. It's also advisable not to force to create such small modules that people maybe is not going to install.

Can you reconsider?

**Comment by alexis-via** [comment]
@pedrobaeza In fact, you were right that we could use the registry in both post_install.py and in post-migration.py (it would not have been possible with pre-migration.py). So I'll change the code to use the registry, and I'll make it inheritable.

**Comment by adrienpeiffer** [comment]
:+1: If CI becomes well

**Comment by lmignon** [comment]
:+1:  Once travis is green

**Comment by pedrobaeza** [comment]
What about my question about the proper hook to check for the existence of the field?

**Comment by alexis-via** [comment]
@pedrobaeza Are u happy now ? I added a hook.

**Comment by alexis-via** [comment]
@pedrobaeza Are you OK with my hook ? Can we merge ?

**Comment by alexis-via** [comment]
Travis is green, so we are ready to merge !

**Comment by alexis-via** [comment]
@pedrobaeza OK to merge this one ?

**Comment by pedrobaeza** [comment]
Yeah, sure.

:+1:

**Pull request merged**